hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -665,13 +665,25 @@ public final Stream<FileSlice> getLatestFileSlicesBeforeOrOn(String partitionStr
       readLock.lock();
       String partitionPath = formatPartitionKey(partitionStr);
       ensurePartitionLoadedCorrectly(partitionPath);
-      Stream<FileSlice> fileSliceStream = fetchLatestFileSlicesBeforeOrOn(partitionPath, maxCommitTime)
-          .filter(slice -> !isFileGroupReplacedBeforeOrOn(slice.getFileGroupId(), maxCommitTime));
+      Stream<Stream<FileSlice>> allFileSliceStream = fetchAllStoredFileGroups(partitionPath)
+              .map(fg -> fg.getAllFileSlicesBeforeOn(maxCommitTime).filter(slice -> !isFileGroupReplacedBeforeOrOn(slice.getFileGroupId(), maxCommitTime)));
       if (includeFileSlicesInPendingCompaction) {
-        return fileSliceStream.map(this::filterBaseFileAfterPendingCompaction).map(this::addBootstrapBaseFileIfPresent);
+        return allFileSliceStream
+                .map(sliceStream ->
```

Comment:
We should make this change inside `filterBaseFileAfterPendingCompaction` itself

Reply:
`filterBaseFileAfterPendingCompaction` is a map function, seems we can't make this change with map function

Reply:
I still think it would be better to change in `filterBaseFileAfterPendingCompaction` otherwise we will have to apply the fix at other callers of this method as well. 

You can change the method to accept a stream and return a stream, and then at the callers do something like:
```
filterBaseFileAfterPendingCompaction(fileSliceStream).map(this::addBootstrapBaseFileIfPresent)
```

Reply:
@microbearz we can do it even simpler: convert `filterBaseFileAfterPendingCompaction` to return `Stream` and replace `map` w/ `flatMap`